spec/extensions/forbid_lazy_load_spec.rb

```diff
@@ -15,312 +16,364 @@
-    proc{@o1.t(:forbid_lazy_load=>true, :reload=>true)}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
-    proc{@o1.ts(:forbid_lazy_load=>true, :reload=>true)}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
-    proc{@o1.mtm_ts(:forbid_lazy_load=>true, :reload=>true)}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
-    proc{@o1.otoo_t(:forbid_lazy_load=>true, :reload=>true)}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
-    proc{@o1.oto_t(:forbid_lazy_load=>true, :reload=>true)}.must_raise Sequel::Plugins::ForbidLazyLoad::Error
-  end
+    include model_instance_specs
 
-  it "should not effect naked datasets" do
-    @c.naked.all.must_equal [{:id=>2, :t_id=>3}]
-    @c.naked.where_each(:id=>1){|x| break x}.must_equal(:id=>2, :t_id=>3)
-  end
+    it "should allow forbid lazy load if retrieved by dataset via Dataset#all" do
+      o = @c.all.first
+      o.allow_lazy_load.must_be_same_as o
+      o.t.must_equal @o2
+      o.ts.must_equal [@o2]
+      o.mtm_ts.must_equal [@o2]
+      o.otoo_t.must_equal @o2
+      o.oto_t.must_equal @o2
+    end
 
-  it "should handle datasets without row_procs" do
-    ds = @c.naked
-    ds.all.first.must_equal(:id=>2, :t_id=>3)
-    ds.each{|x| break x}.must_equal(:id=>2, :t_id=>3)
-    ds.where_each(:id=>1){|x| break x}.must_equal(:id=>2, :t_id=>3)
-    ds.first(2)[0].must_equal(:id=>2, :t_id=>3)
-    ds.first(2){id > 0}[0].must_equal(:id=>2, :t_id=>3)
-    ds.first.must_equal(:id=>2, :t_id=>3)
-    ds.first{id > 1}.must_equal(:id=>2, :t_id=>3)
-    ds.first(:id=>2).must_equal(:id=>2, :t_id=>3)
-    ds.with_pk(1).must_equal(:id=>2, :t_id=>3)
-  end
+    it "should allow lazy load if retrieved by dataset via Dataset#each" do
+      o = @c.each{|x| break x}
+      o.allow_lazy_load.must_be_same_as o
+      o.t.must_equal @o2
+      o.ts.must_equal [@o2]
+      o.mtm_ts.must_equal [@o2]
+      o.otoo_t.must_equal @o2
+      o.oto_t.must_equal @o2
+    end
 
-  it "should handle datasets with row_procs different from the model" do
-    ds = @c.dataset.with_row_proc(proc{|x| x})
-    ds.all.first.must_equal(:id=>2, :t_id=>3)
-    ds.each{|x| break x}.must_equal(:id=>2, :t_id=>3)
-    ds.where_each(:id=>1){|x| break x}.must_equal(:id=>2, :t_id=>3)
-    ds.first(2)[0].must_equal(:id=>2, :t_id=>3)
-    ds.first(2){id > 0}[0].must_equal(:id=>2, :t_id=>3)
-    ds.first.must_equal(:id=>2, :t_id=>3)
-    ds.first{id > 1}.must_equal(:id=>2, :t_id=>3)
-    ds.first(:id=>2).must_equal(:id=>2, :t_id=>3)
-    ds.with_pk(1).must_equal(:id=>2, :t_id=>3)
+    it "should allow lazy load if retrieved by dataset via Dataset#where_each" do
+      5.times do
+        o = @c.where_each(:id=>1){|x| break x}
+        o.allow_lazy_load.must_be_same_as o
+        o.t.must_equal @o2
+        o.ts.must_equal [@o2]
+        o.mtm_ts.must_equal [@o2]
+        o.otoo_t.must_equal @o2
+        o.oto_t.must_equal @o2
+      end
+    end
```

Comment:
Along with `include model_instance_specs`, I just inverted a couple of the core data set specs for this group. I didn't copy all of them because it feels too verbose/duplicative. This should be enough to prove the association behavior is working as expected.